### ISSUE TITLE
ELM in JAX

### ISSUE THREAD

**Issue by ArtemGr** [ArtemGr/bounty#35]
Should implement the simple/original ELM as a Python module (with PyO3),
in order
a) to reuse the SVD/pinv which are readily available in Python (vs link speed costs in Rust),
b) to make it easier reusing or testing AI experiments from Python notebooks and such.

p.s. Of note is https://github.com/PyO3/rust-numpy, should benchmark how fast can we make a numpy array with it.
Should also measure the overhead of Python code. If the majority of time is spent in the [`pinv`](https://github.com/numpy/numpy/blob/v1.22.0/numpy/linalg/linalg.py#L1903-L1999) then building the hidden layer in Rust will not give us that much of a boost, though feeding `ndarray` to `pinv` might help.
p.s. Measurement shows that `build_h` takes most of the time.
```
train] build_h: 71.754
train] pinv: 0.1656
train] matmul: 0.0
```
For the native version having the rows of the `inputs` list as [ndarray](https://numpy.org/doc/stable/reference/arrays.ndarray.html) should also help. (Converting the entire `inputs` to `ndarray` is likely a waste of time though).

p.s. [Might be a relevant note](https://stackoverflow.com/a/63820102/257568) about `ndarray` random access.

p.s. A better option might be to generate H with JAX.
Likely useful as a Stargate, to be followed with mutation and gradient descent.

**Comment by ArtemGr** [comment]
```
def build_h (inputs, weights, bias):
  if inputs.shape[1] == 1:
    return jax.vmap (
      lambda x, w, b: jax.nn.silu (jnp.dot (x, w) + b),
      in_axes = (0, None, None)) (
        inputs, weights, bias)
```

The rest is straightforward.

Note, also, that this is not much different from a traditional set of hidden layers. Specifics of the [computing reservoir](https://en.wikipedia.org/wiki/Reservoir_computing) are less important.
